mypy/main.py

```diff
@@ -482,6 +528,48 @@ def add_invertible_flag(flag: str,
         print("Warning: --no-fast-parser no longer has any effect.  The fast parser "
               "is now mypy's default and only parser.")
 
+    # Infer Python version and/or executable if one is not given
```

Comment:
Since this is kinda long, could consider making this a helper function, as

```
try:
    options.python_version, options.python_executable = _infer_version_and_executable(special_opts)
except PythonExecutableInferenceError:
     parser.error(str(e))
     sys.exit(2)
```